columnflow/tasks/framework/mixins.py

```diff
@@ -1390,3 +1395,28 @@ def raise_if_not_finite(cls, ak_array: ak.Array) -> None:
                     f"found one or more non-finite values in column '{route.column}' "
                     f"of array {ak_array}",
                 )
+
+    @classmethod
+    def raise_if_overlapping(cls, ak_arrays: Sequence[ak.Array]) -> None:
+        """
+        Checks whether fields of *ak_arrays* overlap.
+
+        :param ak_arrays: Arrays with fields to check.
+        :raises ValueError: If at least one overlap is found.
+        """
+        from columnflow.columnar_util import get_ak_routes
+
+        # when less than two arrays are given, there cannot be any overlap
+        if len(ak_arrays) < 2:
+            return
+
+        # determine overlapping routes
+        counts = Counter(sum(map(get_ak_routes, ak_arrays), []))
+        overlapping_routes = [r for r, c in counts.items() if c > 1]
+
+        # raise
+        if overlapping_routes:
+            raise ValueError(
+                f"found {len(overlapping_routes)} overlapping columns across {len(ak_arrays)} "
+                f"arrays: {','.join(overlapping_routes)}",
```

Comment:
This isn't really `arrays`, but more `fields` or `routes` or `columns` right? So it might be more appropriate to write sth like

```suggestion
                f"columns: {','.join(overlapping_routes)}",
```

Reply:
Yup! Added in the same commit as the change below.